src/Product/SearchProvider.php

```diff
@@ -210,28 +210,28 @@ public function renderFacets(ProductSearchContext $context, ProductSearchResult
     {
         list($activeFilters, $displayedFacets, $facetsVar) = $this->prepareActiveFiltersForRender($context, $result);
 
-        // No need to render without facets
-        if (empty($facetsVar)) {
-            return '';
-        }
-
-        $this->module->getContext()->smarty->assign(
-            [
-                'show_quantities' => Configuration::get('PS_LAYERED_SHOW_QTIES'),
-                'facets' => $facetsVar,
-                'js_enabled' => $this->module->isAjax(),
-                'displayedFacets' => $displayedFacets,
-                'activeFilters' => $activeFilters,
-                'sort_order' => $result->getCurrentSortOrder()->toString(),
-                'clear_all_link' => $this->updateQueryString(
-                    [
-                        'q' => null,
-                        'page' => null,
-                    ]
-                ),
-            ]
-        );
+        // If there are no facets, initialize empty array.
+        if (empty($facetsVar))
+            $this->module->getContext()->smarty->assign('displayedFacets', []);
+        else 
+            $this->module->getContext()->smarty->assign(
+                [
+                    'show_quantities' => Configuration::get('PS_LAYERED_SHOW_QTIES'),
+                    'facets' => $facetsVar,
+                    'js_enabled' => $this->module->isAjax(),
+                    'displayedFacets' => $displayedFacets,
+                    'activeFilters' => $activeFilters,
+                    'sort_order' => $result->getCurrentSortOrder()->toString(),
+                    'clear_all_link' => $this->updateQueryString(
+                        [
+                            'q' => null,
+                            'page' => null,
+                        ]
+                    ),
+                ]
+            );
```

Comment:
```suggestion
        if (empty($facetsVar)) {
            $this->module->getContext()->smarty->assign('displayedFacets', []);
        } else {
            $this->module->getContext()->smarty->assign(
                [
                    'show_quantities' => Configuration::get('PS_LAYERED_SHOW_QTIES'),
                    'facets' => $facetsVar,
                    'js_enabled' => $this->module->isAjax(),
                    'displayedFacets' => $displayedFacets,
                    'activeFilters' => $activeFilters,
                    'sort_order' => $result->getCurrentSortOrder()->toString(),
                    'clear_all_link' => $this->updateQueryString(
                        [
                            'q' => null,
                            'page' => null,
                        ]
                    ),
                ]
            );
        }
```

Reply:
Should it need to also initialize js_enabled and other.
Maybe 
```suggestion
        $this->module->getContext()->smarty->assign(
            [
                'show_quantities' => Configuration::get('PS_LAYERED_SHOW_QTIES'),
                'facets' => $facetsVar,
                'js_enabled' => $this->module->isAjax(),
                'displayedFacets' => $displayedFacets,
                'activeFilters' => $activeFilters,
                'sort_order' => $result->getCurrentSortOrder()->toString(),
                'clear_all_link' => $this->updateQueryString(
                    [
                        'q' => null,
                        'page' => null,
                    ]
                ),
            ]
        );
```

Or maybe this is enough?

Reply:
I think `displayedFacets` is the minimum required variable to assign to smarty. 

If you refer to the modified tpl file, `$displayedFacets` is checked for count at the beginning; if it is empty then other variables will not be touched, so I don't think it is necessary to include other variables. 

`renderFacets` is called every time the product listing is changed (i.e. a new filter is applied, or cleared filter). So in the case that the facets is not empty, it will have the required variables when the template is rendered anyway.